main.go

```diff
@@ -29,139 +30,221 @@ type backup struct {
 	in_use                 bool
 }
 
-func main() {
-
-	//get configuration options from environment variables
-	dailyBackupsRetained, err := strconv.Atoi(os.Getenv("DAILY_BACKUPS"))
-	if err != nil {
-		log.Fatalf("Error retrieving DAILY_BACKUPS env var: %v", err)
+type backupconfig struct {
+	Name              string `yaml:"name"`
+	KanisterNamespace string `yaml:"kanisterNamespace"`
+	BlueprintName     string `yaml:"blueprintName"`
+	ProfileName       string `yaml:"profileName"`
+	Retention         struct {
+		Backups string `yaml:"backups"`
+		Minutes string `yaml:"minutes"`
+		Hours   string `yaml:"hours"`
+		Days    string `yaml:"days"`
+		Months  string `yaml:"months"`
+		Years   string `yaml:"years"`
 	}
+}
 
-	weeklyBackupsRetained, err := strconv.Atoi(os.Getenv("WEEKLY_BACKUPS"))
+func main() {
+	// creates the in-cluster config
+	config, err := rest.InClusterConfig()
 	if err != nil {
-		log.Fatalf("Error retrieving WEEKLY_BACKUPS env var: %v", err)
-	}
-
-	kanisterNamespace := os.Getenv("KANISTER_NAMESPACE")
-	if len(kanisterNamespace) == 0 {
-		log.Fatalf("KANISTER_NAMESPACE value cannot be empty")
-	}
-
-	blueprintName := os.Getenv("BLUEPRINT_NAME")
-	if len(blueprintName) == 0 {
-		log.Fatalf("BLUEPRINT_NAME value cannot be empty")
-	}
-
-	s3ProfileName := os.Getenv("S3_PROFILE_NAME")
-	if len(s3ProfileName) == 0 {
-		log.Fatalf("S3_PROFILE_NAME value cannot be empty")
+		panic(err.Error())
 	}
 
-	//set env schedule
-	const evalSchedule string = "1/5 * * * *"
-
-	//creates the in-cluster config
-	config, err := rest.InClusterConfig()
+	// creates the out-of-cluster config
```

Comment:
no need to keep commented code!

Reply:
it makes it easier to debug out of cluster, but you are right

Reply:
I've removed this now